lldb/source/Plugins/SymbolFile/DWARF/DWARFDIE.cpp

```diff
@@ -434,6 +442,19 @@ static void GetTypeLookupContextImpl(DWARFDIE die,
     auto push_ctx = [&](CompilerContextKind kind, llvm::StringRef name) {
       context.push_back({kind, ConstString(name)});
     };
+
+    // Since mangled names are unique there's no need to build an entire context.
+    if (use_mangled_name) {
```

Comment:
If all that `use_mangled_name` is doing is to create a vector with 1 entry, which is the mangled name, can we just do so at the callsite of this function, instead of embedding it here?